src/figuresClasses.ts

```diff
@@ -1,11 +1,73 @@
-export interface Figure {}
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
+
+  constructor(
+    public color: Color,
+    public a = 0,
+    public b = 0,
+    public c = 0,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a >= b + c || b >= a + c || c >= a + b) {
```

Comment:
[CODE STYLE] - If you throw an error try to provide its detailed description in the error message. The current message `sides 1, 2 and 3 can't form a triangle` could be more descriptive, stating the conditions that are not met for forming a triangle.